### ISSUE TITLE
Allow TextView formatting 

### ISSUE THREAD

**Issue by punkkeks** [gyscos/Cursive#162]
I'd like to see some bold, italic, underline in my Textview, but currently it is not possible afaik.
I guess all the underlying backends support this?

**Comment by gyscos** [comment]
Yes, styled views is something I'd like to include but is currently missing from Cursive. There are two parts to this:
* Update backends to allow views to print bold/italic/... text
* Update text views to specify the style, or even better parse marked-up text to have mixed styling.

There was some thoughts on the second point in https://github.com/gyscos/Cursive/pull/106, but I realize the first point may not even be ready yet, so I'll start looking into that.

**Comment by gyscos** [comment]
Just pushed 2d7f8295cf91724a4fd7aa22f1e97dccac6ea63c, which brings the `Bold`, `Italic` and `Underline` effects, so it's now at least *possible* to print such text.

The next step will be to update TextView to apply an `Effect` to its content; and the step after that (which will likely take a bit more work) will be to parse marked-up text, and apply various effects to only segments of the text.

**Comment by hellow554** [comment]
Hurray! 😊
I really like the idea to use marked-up text, let's use markdown? ;)
Here I see two possibilities

A. Use a simple regex to parse \**italic*\*, \*\***bold**\*\* or \*\*\****bolditalic***\*\*\* text 
B. Use a library (e.g. [pulldown](https://github.com/google/pulldown-cmark) ) which might be a bit overkill for just those three cases

**Comment by gyscos** [comment]
The first step is to define an intermediate representation (a list of "styled' spans of text), and adapt the current line-breaking algorithm to work on that. I'll probably look into [xi-unicode's](https://docs.rs/xi-unicode/0.1.0/xi_unicode/struct.LineBreakLeafIter.html) for that.

Then, we'll add various parsers from `&str` to this representation, and markdown sounds like a prime choice. [Pulldown](https://docs.rs/pulldown-cmark/0.1.0/pulldown_cmark/struct.Parser.html), looks interesting, with a reasonable API and few dependencies.

The parsing itself may be behind a feature or even in a separate crate, as it's quite independent from the rest of the code.

**Comment by gyscos** [comment]
I started playing with pulldown-cmark. Note that it only supports [vanilla markdown](http://commonmark.org/help/) and therefore does not support underlining text.

It also generates a lot of markup that we won't support (for now or ever), like tables, horizontal rules, etc... without giving access to the source string for those, so we'll need to manually replace them with strings.
